### DISCUSSION TITLE
Entity cant be resolved/found after update

### DISCUSSION THREAD

**Discussion by ZuBB** [mikro-orm/mikro-orm#5096] [Q&A] [answered]
I saw that v6 had been released, so I decided to stay in safe water for a bit and update to the latest v5. But that version gives me an error, so I started to dig. It turned out that the error started to appear in 5.7.13, which is right after the one that I have right now (5.7.12). I looked through both the changelog and git log of the 13th and did not find anything related to my issue. But most probably I overlooked something.

Here is an error
```
MetadataError: Source file './dist/countries/entities/country.entity.ts' not found. Check your 'entitiesTs' option and verify you have 'compilerOptions.declaration' enabled in your 'tsconfig.json'. If you are using webpack, see https://bit.ly/35pPDNn
```

and here is my `mikro-orm.config.ts` file
```
import { MikroOrmModuleOptions as Options } from '@mikro-orm/nestjs';
import { TsMorphMetadataProvider } from '@mikro-orm/reflection';

const config: Options = {
  type: 'sqlite',
  dbName: 'reard3.sqlite.db',

  entitiesTs: ['./src/**/*.entity.ts'],
  entities: ['./dist/**/*.entity.js'],
  migrations: {
    path: './dist/migrations',
    pathTs: './src/migrations',
  },

  forceUtcTimezone: true,
  metadataProvider: TsMorphMetadataProvider,
  registerRequestContext: false,

  debug: true,
};

export default config;
```

**Comment by B4nan** [comment]
The config looks good, can't help without a complete reproduction.

**Reply by ZuBB** [reply]
Thanks for your reply, Martin. I will cook a repro or share a repo in some form. But I need a bit of time for that

**Reply by ZuBB** [reply, marked as answer]
So, I have solved my issue. I still do not have a complete understanding of it, but this kind of thing may happen.

So, I am using nest v10 + mikro-orm v5.7. Both back and front are in the same repo through monorepo/npm workspaces.

Nest docs claim they support SWC. To enable it, you need to add two options to `nest-cli.json.` which I did. The thing is that this recipe is valid for regular cases. For the monorepo you need to do other magic (all the details can be found in nest docs). 

How this used to work for monorepo and why it stopped with an update of mikro-orm from one patch version to another is a puzzle that I am not sure I want to solve. Hope this will save someone some time

**Reply by B4nan** [reply]
Well, I don't think SWC will give you `.d.ts` files, and those are needed for ts-morph to work (in production mode, so when you execute compiled files, which I believe is how nest currently works, they don't execute the TS files directly, e.g. as we do with ts-node).
